Neos.Neos/Classes/Controller/Service/ContentDimensionsController.php

```diff
@@ -76,4 +77,73 @@ public function indexAction()
             );
         }
     }
+
+    /**
+     * Returns only presets of the dimension specified by $dimensionName. But even though only one dimension is returned,
+     * the output follows the structure as described in ContentDimensionPresetSourceInterface::getAllPresets().
```

Comment:
the `ContentDimensionPresetSourceInterface` doesn't exist in 9.0

only in 8.x https://github.com/neos/neos-development-collection/blob/463c1503ef3f39cdf79d09111ce076038ab90262/Neos.ContentRepository/Classes/Domain/Service/ContentDimensionPresetSourceInterface.php#L30-L51


i guess you copied it from this outdated comment: https://github.com/neos/neos-development-collection/blob/a4658c1e99cf2d7499806478b76bc50e546210b7/Neos.Neos/Classes/Controller/Service/ContentDimensionsController.php#L54 (which needs to be adjusted as well ^^)

Reply:
I'm unsure what to do here :/

I restored the `show` action from it's old state in 8.3 and left the method comment as it was. The comment of the `index` action is wrong in more than one way btw (The method does not produce JSON, unless told to do so by the request format).

I'm struggling to find a better description. The `show` action tries to keep a stable interface to the outside. This means that we need to keep a parameter name like `chosenDimensionPresets`, which has no meaning to the new ESCR (It would now rather be `referenceDimensionSpacepoint` or something).

Since it seems to me that this entire Controller now only serves as a legacy-compatibility-glue thing for the UI, I would actually like to remove it entirely and establish a proper replacement in `Neos.Neos.Ui`.

Reply:
Hmm yes so it seems ;) so we should add this to our big improvement board but keep your changes as is for now ;)

Reply:
Do you want to add any changes to this MR or are we ready to merge this for now? We have quite some dangling MRs which are needed by #3569.

Reply:
Imho this can be merged as is. Just needs a second approval :)